<launch>
 
 <include file="$(find point_cloud_image_creator)/launch/point_cloud_image_creator.launch" >
   <arg name="input" value="/cluster_decomposer_final/debug_output" />
 </include>
   
 <arg name="input_cloud" default="/camera/depth_registered/points" />
 <arg name="input_mask" default="/cloud_image/output/image" />
 <arg name="input_pmap" default="/fcn_object_detector/pmap" />

 <node pkg="fcn_object_detector" type="fcn_point_map_node"
       name="fcn_point_map_node" output="screen" >
   <remap from="input_cloud" to="$(arg input_cloud)" />
   <remap from="input_mask" to="$(arg input_mask)" />
   <remap from="input_pmap" to="$(arg input_pmap)" />

   <remap from="input_model" to="/multi_plane_estimate/output_refined_coefficients" />
 </node>

 <node pkg="jsk_pcl_ros" type="cluster_point_indices_decomposer"
       name="fcn_segmentation_decomposer" >
   <remap from="~input" to="output/points" />
   <remap from="~target" to="output/indices" />
  
   <remap from="~align_planes" to="/multi_plane_estimate/output_refined_polygon" />
   <remap from="~align_planes_coefficients" to="/multi_plane_estimate/output_refined_coefficients" />
   <param name="align_boxes" value="true" />
   <param name="use_pca" value="true" />
   <param name="publish_clouds" value="true" />
   <param name="publish_tf" value="true" />
 </node>

 
 
</launch>
